cmd/retrolabeler/main.go

```diff
@@ -0,0 +1,332 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gobwas/glob"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/schollz/progressbar/v3"
+	"github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+	"gopkg.in/yaml.v3"
+)
+
+type Label struct {
+	Name string
+	All  []glob.Glob
+	Any  []glob.Glob
+}
+
+type matchObject struct {
+	Any []string `yaml:"omitempty"`
+	All []string `yaml:"omitempty"`
+}
+
+func parseGlob(val string, label string) (globby glob.Glob, err error) {
+	if globby, err = glob.Compile(val); err != nil {
+		log.Error().Msgf("Failed to parse glob \"%v\" for label %v, skipped", val, label)
+	}
+	return
+}
+
+func parseGlobArray(vals []string, label string) (globs []glob.Glob, err error) {
+	for _, str := range vals {
+		var globby glob.Glob
+		if globby, err = parseGlob(str, label); err != nil {
+			return
+		}
+		globs = append(globs, globby)
+	}
+	return
+}
+
+func ParseLabelerConfig() ([]Label, error) {
+	configBytes, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		log.Fatal().Msgf("Failed to read from stdin: %v", err)
+		return nil, err
```

Comment:
This will never happen because you call `Fatal`, which will call `os.Exit(1)`. I would suggest avoiding `Fatal`  anywhere except `main` itself, because it kills the program immediately, and will ignore all `defer` statements.